### PR TITLE
[4.0] Language library prepared statements

### PR DESCRIPTION
**Pull request by SharkyKZ**
### Summary of Changes

Adds some prepared statements and cleans up queries in `Joomla\CMS\Language` namespace.

### Testing Instructions

Set up a multilingual site, install multilingual sample data.
View an article in frontend, check that language switcher shows correct associations.

### Expected result

Works like before.

### Documentation Changes Required

No.

### CODE REVIEW

File: libraries/src/Language/Associations.php [hunk 1]
```diff
@@ -85,8 +105,10 @@ public static function getAssociations($extension, $tablename, $context, $id, $p
 			{
 				$query->join(
 					'INNER',
-					$db->quoteName('#__categories', 'ca') . ' ON ' . $db->quoteName('c2.' . $catField) . ' = ca.id AND ca.extension = ' . $db->quote($extension)
+					$db->quoteName('#__categories', 'ca'),
+					$db->quoteName('c2.' . $catField) . ' = ' . $db->quoteName('ca.id') . ' AND ca.extension = :extension2'
```
Comment (Quy) [inline libraries/src/Language/Associations.php]:
```suggestion
					$db->quoteName('c2.' . $catField) . ' = ' . $db->quoteName('ca.id') . ' AND ' . $db->quoteName('ca.extension') . ' = :extension2'
```

File: libraries/src/Language/Associations.php [hunk 2]
```diff
@@ -85,8 +105,10 @@ public static function getAssociations($extension, $tablename, $context, $id, $p
...
 				)
+					->bind(':extension2', $extension)
 					->select(
 						$query->concatenate(
 							array('ca.id', 'ca.alias'),
```
Comment (Quy) [inline libraries/src/Language/Associations.php]: Apply `quoteName`

File: libraries/src/Language/Associations.php [hunk 3]
```diff
@@ -46,21 +47,40 @@ public static function getAssociations($extension, $tablename, $context, $id, $p
...
-			$categoriesExtraSql = (($tablename === '#__categories') ? ' AND c2.extension = ' . $db->quote($extension) : '');
-			$query = $db->getQuery(true)
-				->select($db->quoteName('c2.language'))
+			$db                 = Factory::getDbo();
+			$query              = $db->getQuery(true);
+			$categoriesExtraSql = '';
+
+			if ($tablename === '#__categories')
+			{
+				$categoriesExtraSql = (' AND c2.extension = :extension1');
```
Comment (Quy) [inline libraries/src/Language/Associations.php]:
```suggestion
				$categoriesExtraSql = ' AND c2.extension = :extension1';
```

**Comment by wilsonge** [comment]
Thanks!

**Pull request merged**